hub/core/chunk_engine.py

```diff
@@ -543,6 +546,25 @@ def append(self, sample: SampleValue):
         """Formats a single `sample` (compresseses/decompresses if applicable) and feeds it into `_append_bytes`."""
         self.extend([sample])
 
+    def clear(self):
+        """Deletes all samples."""
+        self.cache.check_readonly()
+
+        n_samples = self.num_samples
+        chunk_names = self.get_chunk_names_for_multiple_indexes(0, n_samples, n_samples)
+        commit_id = self.version_state["commit_id"]
+        chunk_keys = [get_chunk_key(self.key, name, commit_id) for name in chunk_names]
+
+        enc_key = get_chunk_id_encoder_key(self.key, commit_id)
+        enc = ChunkIdEncoder()
+        self.cache[enc_key] = enc
+
+        for key in chunk_keys:
+            del self.cache[key]
```

Comment:
How is synchronization handled with multiple clients?  E.g. more than one client is trying to clear the same tensor.

Reply:
Datasets are locked against simultaneous changes so this would not be possible.